Apps/W1/EDocumentConnectors/SignUp/app/src/APIRequests.Codeunit.al

```diff
@@ -0,0 +1,221 @@
+        exit(SendRequest(HttpRequestMessage, HttpResponseMessage));
+    end;
+
+    procedure GetMarketPlaceCredentials(var HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage): Boolean
+    var
+        ConnectionSetup: Record ConnectionSetup;
+        Auth: Codeunit Auth;
+        BaseUrlTxt: Label '%1/api/Registration/init?EntraTenantId=%2', Locked = true;
+    begin
+        InitRequest(ConnectionSetup, HttpRequestMessage, HttpResponseMessage);
+        HttpRequestMessage := PrepareRequestMsg("Http Request Type"::POST, StrSubstNo(BaseUrlTxt, Auth.GetRootUrl(), Auth.GetBCInstanceIdentifier()));
+        exit(SendRequest(HttpRequestMessage, HttpResponseMessage, true));
+    end;
+
+    local procedure InitRequest(var ConnectionSetup: Record ConnectionSetup; var HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage)
+    begin
+        Clear(HttpRequestMessage);
+        Clear(HttpResponseMessage);
+        if not ConnectionSetup.Get() then
+            Error(MissingSetupErr);
+    end;
+
+    local procedure SendRequest(HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage): Boolean
+    begin
+        SendRequest(HttpRequestMessage, HttpResponseMessage, false);
+    end;
+
+    local procedure SendRequest(HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage; RootRequest: Boolean): Boolean
+    var
+        Auth: Codeunit Auth;
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+    begin
+        HttpRequestMessage.GetHeaders(HttpHeaders);
+        if RootRequest then
+            HttpHeaders.Add('Authorization', Auth.GetRootBearerAuthText())
+        else
+            HttpHeaders.Add('Authorization', Auth.GetBearerAuthText());
+        exit(HttpClient.Send(HttpRequestMessage, HttpResponseMessage));
+    end;
+
+    local procedure PrepareRequestMsg(HttpRequestType: Enum "Http Request Type"; Uri: Text) RequestMessage: HttpRequestMessage
+    var
+        Headers: HttpHeaders;
+    begin
+        RequestMessage.Method(Format(HttpRequestType));
+        RequestMessage.SetRequestUri(Uri);
+        RequestMessage.GetHeaders(Headers);
+        Headers.Add('Accept', '*/*');
+    end;
+
+    local procedure XmlToTxt(var TempBlob: Codeunit "Temp Blob"): Text
+    var
+        XMLDOMManagement: Codeunit "XML DOM Management";
+        InStr: InStream;
+        Content: Text;
+    begin
+        TempBlob.CreateInStream(InStr, TextEncoding::UTF8);
+        XMLDOMManagement.TryGetXMLAsText(InStr, Content);
+        exit(Content);
+    end;
+
+    local procedure GetDocumentType(EDocument: Record "E-Document"): Text
+    begin
+        if EDocument.Direction = EDocument.Direction::Incoming then
+            exit('ApplicationResponse');
+
+        case EDocument."Document Type" of
```

Comment:
Probably error out if type is not supported. Otherwise request is sent with empty doc type